### PR TITLE
Parallelize page fetch queries

### PR DESCRIPTION
**Pull request by SaurusXI**
Parallelizes page queries according to #354

### CODE REVIEW

File: howdoi/howdoi.py [hunk 1]
```diff
@@ -332,11 +335,7 @@ def _get_questions(links):
     return [link for link in links if _is_question(link)]
 
 
-def _get_answer(args, links):
-    link = get_link_at_pos(links, args['pos'])
-    if not link:
-        return False
-
+def _get_answer(link, args):
```
Comment (gleitz) [inline howdoi/howdoi.py:338]: A lot of the `howdoi` functions (`_parse_cmd`, `_format_output`, etc) take `args` as the first argument so probably best to keep the same here.

File: howdoi/howdoi.py [hunk 2]
```diff
@@ -410,29 +409,41 @@ def _get_answers(args):
     if not question_links:
         return False
 
-    answers = []
-    initial_position = args['pos']
-    multiple_answers = (args['num_answers'] > 1 or args['all'])
+    initial_position = args['pos'] - 1
+    final_position = initial_position + args['num_answers']
+    question_links = question_links[initial_position: final_position]
```
Comment (gleitz) [inline howdoi/howdoi.py:414]: I think you can remove the whitespace between `initial_position: final_position`

File: howdoi/howdoi.py [hunk 3]
```diff
@@ -410,29 +409,41 @@ def _get_answers(args):
...
-        if not args['link'] and not args['json_output'] and multiple_answers:
-            answer = ANSWER_HEADER.format(link, answer, STAR_HEADER)
-        answer += '\n'
-        answers.append({
-            'answer': answer,
-            'link': link,
-            'position': current_position
-        })
+    for idx, _ in enumerate(answers):
+        answers[idx]['position'] = idx + 1
```
Comment (gleitz) [inline howdoi/howdoi.py:420]: This is clever, I was wondering how you were doing to keep them in order.

File: howdoi/howdoi.py [hunk 4]
```diff
@@ -410,29 +409,41 @@ def _get_answers(args):
...
 
     return answers
 
 
+def _get_answer_worker(link, args):
```
Comment (gleitz) [inline howdoi/howdoi.py:425]: I suppose this can also take `args` as the first parameter.

**Review by gleitz — APPROVED** [review]
Just a few nits and then this looks good to merge.

File: howdoi/howdoi.py (hunk 3 above)
Comment (SaurusXI) [reply howdoi/howdoi.py:420]: Yeah the threadpool maps elements one-one so ordering was preserved, but indexing from inside a parallel section is a recipe for disaster - so I had to resort to assigning position labels after the fact :smile:

File: howdoi/howdoi.py (hunk 1 above)
Comment (SaurusXI) [reply howdoi/howdoi.py:338]: Right, will push a commit for these changes in a bit

File: howdoi/howdoi.py (hunk 2 above)
Comment (SaurusXI) [reply howdoi/howdoi.py:414]: I think I'll also change the variable names, line 414 seems a little too wordy

**Comment by SaurusXI** [comment]
Push changes according to the nits. Please let me know if any more changes are needed
